File: Mathematics/PalindromeOne.cpp
#include<bits/stdc++.h>
using namespace std;

bool isPal(int n) {
    int temp = n;
    int rev = 0;
    while(temp != 0) {
        int x = temp%10;
        rev = rev*10 + x;
        temp=temp/10;
    }
    return (rev == n);
}

int main() {
    int n; cin>>n;

    bool res = isPal(n);
    cout<<res<<endl;

    return 0;
}

File: src/mathematics/PalindromeOne.java
package mathematics;

import java.util.Scanner;

public class PalindromeOne {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        boolean res = isPal(n);
        System.out.println(res);
    }
    static boolean isPal(int n) {
        int rev = 0;
        int temp = n;
        while(temp != 0) {
            int x = temp%10;
            rev = rev*10 + x;
            temp/=10;
        }
        return (n==rev);
    }
}
